autopush/db.py

```diff
@@ -59,17 +60,31 @@ def create_rotating_message_table(prefix="message", read_throughput=5,
 
 
 def get_rotating_message_table(prefix="message", delta=0):
-    """Gets the message table for the current month.
-
-    This requires the table to already exist or errors will occur.
-
-    """
-    return Table(make_rotating_tablename(prefix, delta))
+    """Gets the message table for the current month."""
+    db = DynamoDBConnection()
+    dblist = db.list_tables()["TableNames"]
+    tablename = make_rotating_tablename(prefix, delta)
+    if tablename not in dblist:
+        return create_rotating_message_table(prefix=prefix, delta=delta)
+    else:
+        return Table(tablename)
 
 
 def create_router_table(tablename="router", read_throughput=5,
                         write_throughput=5):
-    """Create a new router table"""
+    """Create a new router table
+
+    The last_connect index is a value used to determine the last month a user
+    was seen in. To prevent hot-keys on this table during month switchovers the
+    key is determined based on the following scheme:
+
+        (YEAR)(MONTH)(DAY)(HOUR)(0001-0010)
+
+    Note that the random key is only between 1-10 at the moment, if the key is
+    still too hot during production the random range can be increased at the
+    cost of additional queries during GC to locate expired users.
```

Comment:
Using a much larger r+ without the DD/HH would work fine, but I thought it'd be useful to retain the DD/HH knowledge since we could utilize that later to determine how long into the new month it takes on average for people to reconnect, etc. No use for it right now, but maybe later....